knife/lib/chef/knife/core/gem_glob_loader.rb

```diff
@@ -45,22 +45,13 @@ def gem_and_builtin_subcommands
           find_subcommands_via_dirglob
         end
 
-        def find_subcommands_via_dirglob
-          # The "require paths" of the core knife subcommands bundled with chef
-          files = Dir[File.join(Chef::Util::PathHelper.escape_glob_dir(File.expand_path("../../knife", __dir__)), "*.rb")]
-          subcommand_files = {}
-          files.each do |knife_file|
-            rel_path = knife_file[/#{CHEF_ROOT}#{Regexp.escape(File::SEPARATOR)}(.*)\.rb/, 1]
-            subcommand_files[rel_path] = knife_file
-          end
-          subcommand_files
-        end
-
         def find_subcommands_via_rubygems
           files = find_files_latest_gems "chef/knife/*.rb"
+          version_file_match = /#{Regexp.escape(File.join('chef', 'knife', 'version'))}$/
           subcommand_files = {}
           files.each do |file|
-            rel_path = file[/(#{Regexp.escape File.join('chef', 'knife', '')}.*)\.rb/, 1]
+
+            rel_path = file[/(.*)(#{Regexp.escape File.join('chef', 'knife', '')}.*)\.rb/, 2]
```

Comment:
:eyes: This relative path change waa needed to keep everything working well with the relative path change.   I've tested the same change against master to verify it doesn't otherwise cause unexpected problems.